### PR TITLE
nixos/dockerTools: fix includeStorePaths when enableFakechroot

### PR DESCRIPTION
**Pull request by r-k-b**
## Description of changes

After #268458, when setting `enableFakechroot = true` and `includeStorePaths = false`, some of the store paths were getting included into the image anyway, thru `bind-paths`. This resulted in unexpectedly large images.

Now, the images will not contain any store paths under those circumstances.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by roberth — CHANGES_REQUESTED** [review]
Could you add a tests cases for such duplication in `nixos/tests/docker-tools.nix`?

File: pkgs/build-support/docker/default.nix [hunk 1]
```diff
@@ -922,6 +922,7 @@ rec {
                   --sort name \
                   --exclude=./proc \
                   --exclude=./sys \
+                  ${optionalString (!includeStorePaths) "--exclude=.${builtins.storeDir}"} \
```
[the remark this answers is not in the document]
Comment (r-k-b) [reply pkgs/build-support/docker/default.nix]: I've added a test, and fixed my breaking typo.

Hopefully the use of `pkgs.crane` in the tests is ok, I had trouble finding a concise way of listing out the image files.

File: pkgs/build-support/docker/default.nix [hunk 2]
```diff
@@ -922,6 +922,7 @@ rec {
                   --sort name \
                   --exclude=./proc \
                   --exclude=./sys \
+                  --exclude=.${builtins.storeDir} \
```
Comment (roberth) [inline pkgs/build-support/docker/default.nix:925]: Removing this again doesn't fail the test.

Comment (r-k-b) [reply pkgs/build-support/docker/default.nix:925]: I think this is ready to go; is there any other feedback you'd give me?

**Comment by WxNzEMof** [comment]
I have run into the same problem and can confirm that this pull request fixes it. Thank you!

Ping @roberth
CC @Mic92

File: pkgs/build-support/docker/default.nix (hunk 2 above)
Comment (roberth) [reply pkgs/build-support/docker/default.nix:925]: The test fails and succeeds as it should now :+1:

**Review by roberth — APPROVED** [review]

**Comment by WxNzEMof** [comment]
@roberth Kind ping, I think this is ready to go in.

**Comment by roberth** [comment]
@WxNzEMof thanks!

**Pull request merged**